packages/babel-plugin-transform-classes/src/vanilla.js

```diff
@@ -163,10 +163,24 @@ export default class ClassTransformer {
     //
     body.push(t.returnStatement(t.cloneNode(this.classRef)));
 
+    const parent = this.path.findParent(
```

Comment:
ive left comment regarding this [here](https://github.com/babel/babel/pull/7411#discussion_r171531339), it got hidden by github though (i think i might have commented on some older commit)